cassandra/src/main/scala/filodb.cassandra/columnstore/CassandraColumnStore.scala

```diff
@@ -148,6 +148,34 @@ extends ColumnStore with CassandraChunkSource with StrictLogging {
     }
   }
 
+  /**
+    * Reads chunks by querying partitions by ingestion time range and subsequently filtering by user time range.
+    * ** User/Ingestion End times are exclusive **
+    */
+  def getChunksByIngestionTimeRange(datasetRef: DatasetRef,
+                                    splits: Iterator[ScanSplit],
+                                    ingestionTimeStart: Long,
+                                    ingestionTimeEnd: Long,
+                                    userTimeStart: Long,
+                                    userTimeEnd: Long,
+                                    batchSize: Int): Observable[Seq[RawPartData]] = {
+    val partKeys = Observable.fromIterator(splits).flatMap {
+      case split: CassandraTokenRangeSplit =>
+        val indexTable = getOrCreateIngestionTimeIndexTable(datasetRef)
+        logger.debug(s"Querying cassandra for partKeys for split=$split")
+        indexTable.scanPartKeysByIngestionTime(split.tokens, ingestionTimeStart, ingestionTimeEnd)
+      case split => throw new UnsupportedOperationException(s"Unknown split type $split seen")
+    }
+
+    import filodb.core.Iterators._
+
+    val chunksTable = getOrCreateChunkTable(datasetRef)
+    partKeys.bufferTimedAndCounted(1.second, batchSize).map { parts =>
+      logger.debug(s"Querying cassandra for chunks from ${parts.size} partitions")
+      chunksTable.readRawPartitionRangeBB(parts, userTimeStart, userTimeEnd).toIterator().toSeq
```

Comment:
This is going to serially read one partition at a time.  I know there is parallelism in Spark job, but I think this will be too slow in practice.   At least, let's add a TODO here and revisit.  I think you will actually need to group partitions to read in bulk, which is much faster.